core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -256,6 +265,9 @@ public synchronized void resetConnection() {
   }
 
   /**
+   * Returns the address of the remote. This method may be overridden to change the address in case
+   * of remote failover.
```

Comment:
All sites which access `mAddress` require holding the lock on the `AbstractClient` instance